controllers/ArticleController.js

```diff
@@ -612,6 +605,80 @@ class ArticleController {
 
     return res.status(200).json({ status: 201, ...reports, pages });
   }
+
+  /**
+   * @author Chris
+   * @param {Object} req
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async getFeed(req, res){
+    const { currentUser } = req;
+    const { page = 1 } = req.query;
+    const limit = 10;
+    const offset = limit * (page - 1);
+
+    const following = await Follow.findAll({ where: { follower: currentUser.id }, attributes: ['followee'] });
+    const filteredFollowing = await following.map(a => a.followee);
+
+    const reads = await Reader.findAll({ where: { userId: currentUser.id }, attributes: ['articleId'], include: [{
+      model: Article,
+      as: 'article',
+      attributes: ['title', 'tagList'],
+    }] });
+    
+    const filteredReads = await reads.filter( r => r.article.tagList);
+    const tags = [];
+    if (filteredReads) {
+      for(let i = 0; i < filteredReads.length; i +=1){
+        tags.push(...filteredReads[i].article.tagList);
+      }      
+    }
+
+    let articles = await Article.findAndCountAll({ 
+      where: 
+      {
+        [Op.or]: 
+        [
+          { 
+            tagList: {[Op.contained]: tags},
+            status: 'published'
+          },
+          {
+            [Op.or]: [ { userId:  filteredFollowing } ],
+            status: 'published' 
+          }
+        ]
+      },
+      include: [
+        {
+          model: User,
+          as: 'author',
+          attributes: ['username', 'firstName', 'lastName', 'image']
+        }
+      ],
+      offset,
+      limit
+    });
+
+    let pages = Math.ceil(articles.count / limit);
+    if (page > pages) {
```

Comment:
@filschristian You need to return page not found in case the user is trying to fetch a page that does not exist?

Reply:
okay